test.py

```diff
@@ -0,0 +1,136 @@
+from typing import Callable
+import torch
+from PIL import Image
+from networkx import tensor_product
+from utils import metrics
+
+# def test_mse(gt_tensor: torch.tensor, predicted_tensor: torch.tensor):
+#     gt_tensor = torch.randn(100, 100)  # Example ground truth tensor
+#     predicted_tensor = torch.randn(100, 100)  # Example predicted tensor
+#     mse_per_sample
+    
+#     # print('GT tensor : ', gt_tensor)
+#     # print('Predicted tensor : ', predicted_tensor)
+#     assert gt_tensor==predicted_tensor
+
+# def mse(gt_tensor: torch.tensor, predicted_tensor: torch.tensor):
+#     return metrics.mse_per_sample(predicted_tensor,gt_tensor)
+
+def test_mse():
+    # Test case 1
+    gt_tensor_1 = torch.tensor([[[1.0, 2.0], [3.0, 4.0]]])
```

Comment:
- all of these test cases are similar so they are redundant..?
- i think 2 test cases is fine.
- the output should be torch array, not single float

pls add case something like this that tests for only the first image

```
a = np.arange(2*3*4*5).reshape((2, 3, 4, 5))
b = np.zeros((2, 3, 4, 5))
expected_0 = (a[0] - b[0])**2 / (a.shape[1]*a.shape[2]*a.shape[3])
actual_0 = mse_per_sample(a, b)[0]
assert torch.allclose(actual_0, expected_0)

```